src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -122,10 +122,16 @@ private IEnumerable<TypeInferenceInfo> GetTypesSimple(ExpressionSyntax expressio
             }
 
             protected override IEnumerable<TypeInferenceInfo> InferTypesWorker_DoNotCallDirectly(
-                ExpressionSyntax expression)
+                SyntaxNode node)
             {
-                expression = expression.WalkUpParentheses();
-                var parent = expression.Parent;
+                var expression = node as ExpressionSyntax;
+                if (expression != null)
+                {
+                    expression = expression.WalkUpParentheses();
+                    node = expression;
```

Comment:
nit: consider pattern matching: `if (node is ExpressionSyntax expression) node = expression.WalkUpParentheses()`